packages/liveblocks-yjs/src/awareness.ts

```diff
@@ -46,46 +48,93 @@ export class Awareness extends Observable<unknown> {
     this.doc = doc;
     this.room = room;
     this.clientID = doc.clientID;
+    // Add the clientId to presence so we can map it to connectionId later
+    this.room.updatePresence({
+      [Y_PRESENCE_KEY]: { clientID: this.clientID },
+    });
     this.othersUnsub = this.room.events.others.subscribe((event) => {
+      this.rebuildActorToClientMap(event.others);
       // When others are changed, we emit an event that contains arrays added/updated/removed.
       if (event.type === "leave") {
         // REMOVED
         this.emit("change", [
-          { added: [], updated: [], removed: [event.user.connectionId] },
+          {
+            added: [],
+            updated: [],
+            removed: [this.actorToClientMap.get(event.user.connectionId)],
+          },
           "presence",
         ]);
         this.emit("update", [
-          { added: [], updated: [], removed: [event.user.connectionId] },
+          {
+            added: [],
+            updated: [],
+            removed: [this.actorToClientMap.get(event.user.connectionId)],
+          },
           "presence",
         ]);
       }
 
       if (event.type === "enter") {
         // ADDED
         this.emit("change", [
-          { added: [event.user.connectionId], updated: [], removed: [] },
+          {
+            added: [this.actorToClientMap.get(event.user.connectionId)],
+            updated: [],
+            removed: [],
+          },
           "presence",
         ]);
         this.emit("update", [
-          { added: [event.user.connectionId], updated: [], removed: [] },
+          {
+            added: [this.actorToClientMap.get(event.user.connectionId)],
+            updated: [],
+            removed: [],
+          },
```

Comment:
If the user is also using normal Liveblocks presence, and moving their cursor around, generating lots of update events, should we only emit these updates if the `__yjs` key changed here?

```js
{
  "__yjs": { "clientID": 1 },
  "cursor": { "x": 13, "y": 218 }  // 👈 Suppose only this part gets changed a lot
}
```

Reply:
the change event doesn't give us that information right? We'd have to actually deep compare or something.